app/lib/redcarpet/render/html_rouge.rb

```diff
@@ -14,7 +14,11 @@ def link(link, _title, content)
         @options[:link_attributes]&.each do |attribute, value|
           link_attributes += %( #{attribute}="#{value}")
         end
-        %(<a href="#{link}"#{link_attributes}>#{content}</a>)
+        if (/^(http(s*):\/\/)/m.match? link) || link.nil?
```

Comment:
Can I ask why this regexp matches multiple `s` chars and uses `m` (dot matching new lines) as a modifier?

I also suggest replacing `^` with `\A`

Reply:
Ah, I completely forgot `*` goes up to unlimited times and should be limited to 0-1, A bit new to regex. I mixed up `m` and `i` too. Will patch them immediately. I will read up more on testing using RSpec and add tests for the same.

Reply:
Thanks, don't worry, it's all trial and error, that's why we should have some tests especially when regexps are involved :) Let me know if you need help!

Reply:
Is it okay if I let this go stale for a bit? I have university exams coming up which will end in 18 days. RSpec, BDD and the concept of testing are completely new to me. I feel that I will be able to fully focus on specs after exams end and implement them properly.

Reply:
@Amorpheuz since the core functionality is in and the fix is important, would you mind if any of the core contributors takes this to the finish line?

Your name and attribution will still appear in the commit logs.

Let us know!

Reply:
Sure, I don't mind... That was the reason I notified you too about the same :smile:

Reply:
I found that it might be more straight forward if we use `URI.parse` to check for the protocol instead of relying on regex. Something along the line of
```
uri.prepend("//") if URI.parse(link).scheme.nil?
```
how does everybody feel about this?